lib/board.rb

```diff
@@ -34,11 +34,43 @@ def place(ship, coords)
   end
 
   def valid_placement?(ship, coords)
-  
-    coords.find do |coord|
-      !@cells[coord].empty?
-    end
+    same_length?(ship, coords) && is_consecutive_row_or_col(coords) && !is_diagonal?(coords) && ( coords.none? { |coord| is_overlap?(coord) } )
+  end
+
+  def same_length?(ship, coords)
+    ship.length == coords.length
+  end
+
+  def is_diagonal?(coords)
+    !is_row?(coords) && !is_col?(coords)
+  end
+
```

Comment:
Technically, a diagonal coords array has a "consecutive pattern" 
The program doesn't care about diagonal consecutive patterns
So here, diagonal simply means NOT a row OR a column

Reply:
Nice, good explanation.